### ISSUE TITLE
Timeout large files (upstream error in xmlhttprequest?)

### ISSUE THREAD

**Issue by gmaclennan** [philschatz/octokat.js#24]
For large files I am getting an inconsistent uncaught error from xmlhttprequest - I can only guess that it is related to a timeout because sometimes it works, sometimes it doesn't.

See https://travis-ci.org/gmaclennan/ghfs.js/builds/52682152 failing build. Error:

```
/home/travis/build/gmaclennan/ghfs.js/node_modules/octokat/node_modules/xmlhttprequest/lib/XMLHttpRequest.js:208
      && response.headers[header.toLowerCase()]
                 ^
TypeError: Cannot read property 'headers' of undefined
    at getResponseHeader (/home/travis/build/gmaclennan/ghfs.js/node_modules/octokat/node_modules/xmlhttprequest/lib/XMLHttpRequest.js:208:18)
    at /home/travis/build/gmaclennan/ghfs.js/node_modules/octokat/dist/node/request.js:148:38
    at xhr.onreadystatechange (/home/travis/build/gmaclennan/ghfs.js/node_modules/octokat/dist/node/request.js:43:18)
    at dispatchEvent (/home/travis/build/gmaclennan/ghfs.js/node_modules/octokat/node_modules/xmlhttprequest/lib/XMLHttpRequest.js:570:25)
    at setState (/home/travis/build/gmaclennan/ghfs.js/node_modules/octokat/node_modules/xmlhttprequest/lib/XMLHttpRequest.js:589:14)
    at handleError (/home/travis/build/gmaclennan/ghfs.js/node_modules/octokat/node_modules/xmlhttprequest/lib/XMLHttpRequest.js:514:5)
    at ClientRequest.errorHandler (/home/travis/build/gmaclennan/ghfs.js/node_modules/octokat/node_modules/xmlhttprequest/lib/XMLHttpRequest.js:439:14)
    at ClientRequest.emit (events.js:95:17)
    at CleartextStream.socketErrorListener (http.js:1552:9)
    at CleartextStream.emit (events.js:95:17)
```

**Comment by philschatz** [comment]
Can you try and use `#master` for the version of octokat? The defenition of `XMLHttpRequest.getHeaders` changed in `1.7.0` to:

```js
  /**
   * Gets a header from the server response.
   *
   * @param string header Name of header to get.
   * @return string Text of the header or null if it doesn't exist.
   */
  this.getResponseHeader = function(header) {
    if (typeof header === "string"
      && this.readyState > this.OPENED
      && response
      && response.headers
      && response.headers[header.toLowerCase()]
      && !errorFlag
    ) {
      return response.headers[header.toLowerCase()];
    }

    return null;
  };
```

**Comment by gmaclennan** [comment]
xmlhttprequest `1.7.0` seems to fix the problem. `#master` of octokat still references `~1.6.0` in `package.json` though, I changed that to ^1.7.0 to get it working.
Let me know when you publish a release to npm.

**Comment by philschatz** [comment]
Thanks ; ) sorry, I forgot to push earlier. Should be available as `0.4.7`.
